### PR TITLE
Initial setup

### PR DESCRIPTION
**Pull request by nerlichman**
- Added CORS: [rack-cors](https://github.com/cyu/rack-cors)
- Added linters `rubocop` and `rake`.
- Added test suite: `rspec`, `faker`, `factory_bot`
- Added `bullet`
- Added `letter_opener`

### CODE REVIEW

**Review by colembolos — APPROVED** [review]
Alt Text

File: .rubocop.yml [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+  # containing a URI to be longer than Max.
+  AllowURI: true
+  URISchemes:
+    - http
+    - https
+  Exclude:
+    - db/migrate/20161011184702_devise_create_admin_users.rb
+    - db/migrate/20161011151353_devise_create_users.rb
+    - db/migrate/20161027190856_create_delayed_jobs.rb
+    - db/migrate/20181102142200_create_active_storage_tables.active_storage.rb
```
Comment (juan-apa) [inline .rubocop.yml:50]: you can just exclude `db/migrate*`

File: .rubocop.yml [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+
+# Avoid complex methods.
+Metrics/CyclomaticComplexity:
+  Max: 6
+
+Metrics/MethodLength:
+  CountComments: false
+  Max: 24
+  Exclude:
+    - db/migrate/20161011151353_devise_create_users.rb
```
Comment (juan-apa) [inline .rubocop.yml:78]: here too

File: Gemfile [hunk 3]
```diff
@@ -32,13 +32,15 @@ gem 'rack-cors', '~> 1.0', '>= 1.0.6'
 gem 'bootsnap', '>= 1.4.4', require: false
 
 group :development, :test do
+  gem 'bullet', '~> 6.1'
   # Call 'byebug' anywhere in the code to stop execution and get a debugger console
```
Comment (juan-apa) [inline Gemfile:36]: remove these comments from the gemfile.

File: config/environments/development.rb [hunk 4]
```diff
@@ -3,6 +3,16 @@
 require 'active_support/core_ext/integer/time'
 
 Rails.application.configure do
+  config.after_initialize do
+    Bullet.enable        = true
+    Bullet.alert         = true
+    Bullet.bullet_logger = true
+    Bullet.console       = true
+    # Bullet.growl         = true
```
Comment (juan-apa) [inline config/environments/development.rb:11]: remove this comment

**Review by juan-apa — APPROVED** [review]

**Pull request merged**